analyzers/tests/SonarAnalyzer.UnitTest/Extensions/ISymbolExtensionsTest.cs

```diff
@@ -145,13 +147,68 @@ public void IsAutoProperty_NonpropertySymbol_VB()
+    [DataRow("""
+    class Base(int i);
+    class SymbolMember : Base
+    {
+        SymbolMember() : base(1) { }
+    }
+    """, false)]
+    [DataRow("struct SymbolMember();", true)]
+    [DataRow("struct SymbolMember() { }", true)]
+    [DataRow("struct SymbolMember(int a) { }", true)]
+    [DataRow("struct SymbolMember { }", false)]
+    [DataRow("struct SymbolMember(int a) { public SymbolMember() : this(1) { } };", true)]
+    [DataRow("struct SymbolMember { public SymbolMember() { } };", false)]
+    [DataRow("record SymbolMember();", true)]
+    [DataRow("record SymbolMember() { }", true)]
+    [DataRow("record SymbolMember(int a) { }", true)]
+    [DataRow("record SymbolMember { }", false)]
+    [DataRow("record SymbolMember(int a) { SymbolMember() : this(1) { } };", true)]
+    [DataRow("record SymbolMember { SymbolMember() { } };", false)]
+    [DataRow("record struct SymbolMember();", true)]
+    [DataRow("record struct SymbolMember() { }", true)]
+    [DataRow("record struct SymbolMember(int a) { }", true)]
+    [DataRow("record struct SymbolMember { }", false)]
+    [DataRow("record struct SymbolMember(int a) { public SymbolMember() : this(1) { } };", true)]
+    [DataRow("record struct SymbolMember { public SymbolMember() { } };", false)]
+    [DataRow("record class SymbolMember();", true)]
+    [DataRow("record class SymbolMember() { }", true)]
+    [DataRow("record class SymbolMember(int a) { }", true)]
+    [DataRow("record class SymbolMember { }", false)]
+    [DataRow("record class SymbolMember(int a) { SymbolMember() : this(1) { } };", true)]
+    [DataRow("record class SymbolMember { SymbolMember() { } };", false)]
+    [DataRow("readonly struct SymbolMember();", true)]
+    [DataRow("readonly struct SymbolMember() { }", true)]
+    [DataRow("readonly struct SymbolMember(int a) { }", true)]
+    [DataRow("readonly struct SymbolMember { }", false)]
+    [DataRow("readonly struct SymbolMember(int a) { public SymbolMember() : this(1) { } };", true)]
+    [DataRow("readonly struct SymbolMember { public SymbolMember() { } };", false)]
+    public void IsPrimaryConstructor_CS(string code, bool hasPrimaryConstructor)
+    {
+        var typeSymbol = (INamedTypeSymbol)CreateSymbol(code, true, AnalyzerLanguage.CSharp, ParseOptionsHelper.CSharpLatest.First());
+        var methodSymbols = typeSymbol.GetMembers().OfType<IMethodSymbol>();
+
+        methodSymbols.Count(ISymbolExtensionsCS.IsPrimaryConstructor).Should().Be(hasPrimaryConstructor ? 1 : 0);
+    }
+
+    private static ISymbol CreateSymbol(string snippet, bool firstOccurence, AnalyzerLanguage language, ParseOptions parseOptions = null)
     {
-        var (tree, semanticModel) = TestHelper.Compile(snippet, false, language);
-        var node = tree.GetRoot().DescendantNodes().Last(x => x.ToString().Contains(" SymbolMember"));
+        var (tree, semanticModel) = TestHelper.Compile(snippet, false, language, parseOptions: parseOptions);
+        var node = firstOccurence
```

Comment:
This is a very ugly implementation, but I was not able to come up with something more elegant for this problem.

Reply:
You can prepend unwanted names with @ instead:
`[DataRow("readonly struct @SymbolMember(int a) { public SymbolMember() : this(1) { } };")]`